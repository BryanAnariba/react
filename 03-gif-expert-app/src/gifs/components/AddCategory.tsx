import { ChangeEvent, FormEvent, useState } from "react";

interface Props {
  onNewCategory: (value: string) => void
}

export const AddCategory = ({onNewCategory}: Props): JSX.Element => {
  const [category, setCategory] = useState<string>("");

  const addNewCategory = (e: FormEvent<HTMLFormElement>) => {
    e.preventDefault();
    if (category.trim().length !== 0) {
      onNewCategory(category)
    } else {
      onNewCategory('');
    }
  };

  const handleInputChange = (e: ChangeEvent<HTMLInputElement>) => {
    setCategory(e.target.value)
  }

  return (
    <form onSubmit={(e) => addNewCategory(e)}>
      <input
        type="text"
        name="category"
        id="category"
        placeholder="Search gifs"
        value={category}
        onChange={handleInputChange}
      />
      <button type="submit">Add Category</button>
    </form>
  );
};
