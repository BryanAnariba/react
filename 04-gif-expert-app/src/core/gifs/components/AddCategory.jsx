import { useState } from "react";

export const AddCategory = ({onNewCategory}) => {
  const [inputValue, setInputValue] = useState("");

  const onInputChange = (e) => {
    // console.log({onChangeValue: e.target.value});
    setInputValue(e.target.value);
  };

  const onSubmit = (e) => {
    e.preventDefault();

    // console.log('Category Added: ', {inputValue});
    if (inputValue.trim().length <= 1) return;

    // Es como emitir el evento al padre de angular pero emitimos el valor del input al padre
    onNewCategory(inputValue)
    setInputValue('');
  }

  return (
    <form onSubmit={onSubmit}>
      <input
        type="text"
        placeholder="Search Gifs..."
        value={inputValue}
        onChange={onInputChange}
      />
    </form>
  );
};
