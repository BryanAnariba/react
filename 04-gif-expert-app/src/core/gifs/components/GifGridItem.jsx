import PropTypes from 'prop-types';

export const GifGridItem = ({title, url, id}) => {
  return (
    <div className="card">
      <img src={url} alt={title}/>
      <p>{id} - {title}</p>
    </div>
  )
}

GifGridItem.PropTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
  id: PropTypes.string.isRequired,
}