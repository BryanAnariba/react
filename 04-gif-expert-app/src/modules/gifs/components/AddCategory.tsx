import React, { useState } from "react";

interface Props {
  name: string;
  id: string;
  placeholder: string;
  onReceiveCategory: (value: string) => void;
}

export const AddCategory = ({ name, id, placeholder, onReceiveCategory }: Props): JSX.Element => {
  const [inputValue, setInputValue] = useState<string>('');

  const onInputChange = (e: React.ChangeEvent<HTMLInputElement>): void => {
    setInputValue(e.target.value);
  }

  const handleSubmit = (e: React.FormEvent): void => {
    e.preventDefault();
    console.log('======> handle submit executed from AddCategory.test.tsx <======');
    if (inputValue.trim().length !== 0) {
      onReceiveCategory(inputValue.trim().toUpperCase()); // Este es de salida se emite al padre cuando se de enter, peculiar
      setInputValue('');
    }
  }

  return (
    <form onSubmit={handleSubmit} aria-label="form">
      <input type="text" name={name} id={id} placeholder={placeholder} value={inputValue} onChange={onInputChange} />
    </form>
  )
}
