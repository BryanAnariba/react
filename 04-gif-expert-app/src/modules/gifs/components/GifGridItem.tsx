interface Props {
  id: string;
  title: string;
  url: string;
}

export const GifGridItem = ({ id, title, url }: Props): JSX.Element => {
  return (
    <li className="card">
      <p>
        {id} - {title}
      </p>
      <img src={url} alt={'Image ' + title} />
    </li>
  )
}
