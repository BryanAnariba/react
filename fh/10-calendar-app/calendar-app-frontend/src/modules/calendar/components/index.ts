export * from './CalendarEvent';
export * from './CalendarEvent';
export * from './FabAddNew';
export * from './FabDelete';