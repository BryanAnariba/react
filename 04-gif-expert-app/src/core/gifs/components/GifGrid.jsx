import { GifGridItem } from "./GifGridItem";
import { useFetchGifs } from "../hooks";

export const GifGrid = ({ category }) => {
  const { gifs, isLoading } = useFetchGifs(category);
  return (
    <>
      {isLoading && (<h4>Loading Gifs!!!</h4>)}
      <>
          <h4>{category}</h4>
          <ol>
            <div className="card-grid">
              {gifs.map((gif) => (
                <GifGridItem gif={gif} key={gif.id} />
              ))}
            </div>
          </ol>
        </>
    </>
  );
};
