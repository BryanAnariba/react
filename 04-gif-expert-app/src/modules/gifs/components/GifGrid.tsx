import { useGifs } from "../hooks";
import { GifGridItem } from "./GifGridItem";

interface Props {
  category: string;
}

export const GifGrid = ({ category }: Props): JSX.Element => {

  const { gifs, isLoading } = useGifs(category);

  return (
    <>
      <h3>{category}</h3>
      <hr />
      {
        isLoading
          ?
          (
            <div style={{ display: 'flex', justifyContent: 'center', flexDirection: 'row' }}>
              <section style={{ display: 'flex', flexDirection: 'row', justifyContent: 'space-between', verticalAlign: 'middle', gap: '20px' }}>
                <div className="spinner"></div>
                <span style={{margin: 'auto'}}>Loading gifs please wait...</span>
              </section>
            </div>
          )
          :
          (
            <ol className="card-grid">
              {
                gifs.map(gif => <GifGridItem {...gif} key={gif.id} />)
              }
            </ol>
          )
      }
    </>
  )
}
