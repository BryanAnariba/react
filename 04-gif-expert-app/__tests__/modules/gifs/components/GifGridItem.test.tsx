import React from 'react';
import { render } from '@testing-library/react';
import { GifGridItem } from '../../../../src/modules/gifs/components/GifGridItem';

describe('Testing in ./src/modules/gifs/components/GifGridItem.tsx', () => {
  test('Should to be match with the snapshot', () => {
    const { container } = render(<GifGridItem id={'123'} title={'Test title'} url={'Test url'} />);
    expect(container).toMatchSnapshot();
  });
});