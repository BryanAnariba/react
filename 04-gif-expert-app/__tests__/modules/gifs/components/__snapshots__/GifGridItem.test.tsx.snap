// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing in ./src/modules/gifs/components/GifGridItem.tsx Should to be match with the snapshot 1`] = `
<div>
  <li
    class="card"
  >
    <p>
      123
       - 
      Test title
    </p>
    <img
      alt="Image Test title"
      src="Test url"
    />
  </li>
</div>
`;
